### PR TITLE
Fix gaussstretch segfault

### PR DESCRIPTION
**Pull request by chkim-usgs**
## Description

Addresses the `gaussstretch` segfault issue that occurs even when producing a valid output. The related issue below mentions that this issue exists throughout versions 4.1.0-7.2.0. This PR only targets 7.2. How to address the other affected versions?

Also refactors code and converts tests into gtest.

## Related Issue

Only fixes version `7.2` in https://github.com/DOI-USGS/ISIS3/issues/5240

## How Has This Been Validated?

Ran `gaussstretch from=<test.cub> to=<test.output.cub> gsigma=3.0` and confirmed segfault error does not output.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Infrastructure change (changes to things like CI or the build system that do not impact users)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [ ] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/src/base/apps/gaussstretch/gaussstretch.cpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  }
+
+  // Processing routine for the pca with one input cube
+  void gauss(Buffer &in, Buffer &out) {
+    for(int i = 0; i < in.size(); i++) {
+      if(IsSpecial(in[i])) out[i] = in[i];
+      out[i] = stretch[in.Band(i)-1]->Map(in[i]);
+    }
+  }
+}
```
Comment (Kelvinrr) [inline isis/src/base/apps/gaussstretch/gaussstretch.cpp]: We want to avoid these helper functions from entering the Isis namespace. As it could cause problems down the line (e.g. conflicting symbols). Refer to part of the GTest docs about how to refactor these process functions: https://github.com/DOI-USGS/ISIS3/wiki/Writing-ISIS-Tests-Using-Gtest-and-Ctest#process-helper-functions-and-global-variables 

TLDR: use a lambda instead.

File: isis/src/base/apps/gaussstretch/gaussstretch.cpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
+#include "ProcessByLine.h"
+#include "Statistics.h"
+#include "GaussianStretch.h"
+#include "gaussstretch.h"
+
+using namespace std;
+using namespace Isis;
+
+namespace Isis {
+  void gauss(Buffer &in, Buffer &out);
```
Comment (Kelvinrr) [inline isis/src/base/apps/gaussstretch/gaussstretch.cpp]: Delete the forward declaration

File: isis/src/base/apps/gaussstretch/gaussstretch.cpp [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+  vector<GaussianStretch *> stretch;
```
Comment (Kelvinrr) [inline isis/src/base/apps/gaussstretch/gaussstretch.cpp]: This definition can go inside the `gaussstretch(Cube *icube, UserInterface &ui)` as it will be captured by the lambda.

**Review by Kelvinrr — COMMENTED** [review]
Only a two changes left. I think this is good after this.

**Comment by chkim-usgs** [comment]
@Kelvinrr do you have any thoughts about getting this fix into into all previous versions thru 4.1.0? Should I create a PR with the fix for each version?

**Comment by Kelvinrr** [comment]
@chkim-usgs Im pretty sure you shouldnt have to worry about that. As it is a bugfix, whoever pulls the releases will make sure it is ported to the right versions. 

Also (so sorry), one more thing, add a change log update.

**Review by Kelvinrr — APPROVED** [review]
